crates/topos/src/components/node/mod.rs

```diff
@@ -206,28 +238,67 @@ pub(crate) async fn handle_command(
             tokio::select! {
                 _ = signal::ctrl_c() => {
                     info!("Received ctrl_c, shutting down application...");
-                    shutdown(shutdown_trigger, shutdown_receiver).await;
+                    shutdown(basic_controller, shutdown_trigger, shutdown_receiver).await;
                 }
                 Some(result) = processes.next() => {
                     info!("Terminate: {result:?}");
                     if let Err(e) = result {
                         error!("Termination: {e}");
                     }
-                    shutdown(shutdown_trigger, shutdown_receiver).await;
+                    shutdown(basic_controller, shutdown_trigger, shutdown_receiver).await;
                     processes.clear();
                 }
             };
 
             Ok(())
         }
+
+        Some(NodeCommands::Status(status)) => {
+            let mut node_service = NodeService::with_grpc_endpoint(&status.node_args.node);
+            let exit_code = i32::from(!(node_service.call(status).await?));
```

Comment:
What are the possible exit codes here and what do they mean?

Reply:
We didn't modify this, just moved it around. It's a fair question though. @Freyskeyd, do you know?

Reply:
Just FYI, we are going to sunset the way we request the status from a node, so this will change soon. But we can add a comment if you want.

Reply:
The return of service execution is a boolean so the status can be either `0` or `1`